bin/join-metadata-and-clades

```diff
@@ -16,6 +16,10 @@ reference_day = datetime(2020,1,1).toordinal()
 
 column_map = {
     "clade_legacy": "Nextstrain_clade",
+    "clade_nextstrain": "year_letter_clade",
+    "clade_who": "clade_who",
+    "clade_legacy": "clade_legacy",
+    "clade_legacy": "display_clade",
```

Comment:
`year_letter_clade` seems like it weirdly omits that this is the Nextclade/Nextstrain clade… esp. given `clade_who`. What about being consistent with `clade_nextstrain` or similar instead? (I know this gets potentially confusing with `Nextstrain_clade`, but that's already sort of an outlier of a name.)

Instead of `display_clade`, I'd probably just call it `clade`. Assuming that it will continue to include multiple naming systems, e.g. in the style of `23A (Omicron, XBB.1.5)`. Since this is ncov-ingest and we're producing general data files, it doesn't seem appropriate to pre-suppose what column will be used for display by downstream, and the bare `clade` being inclusive of multiple naming systems seems appropriate.

I'd probably drop

```python
    "clade_legacy": "clade_legacy",
```

until we need it, i.e. if/when we change `Nextstrain_clade`. (We might chose to just leave `Nextstrain_clade` alone indefinitely and instead switch ncov to using `display_clade` (or `clade`).) If we _do_ use `clade_legacy`, we might want to consider a more specific name for it than just "legacy".

Reply:
I'd be happy to rename `year_letter_clade` to `clade_nextstrain` as long as it doesn't seem too likely to be confusing. 

For the other things, more discussion on the long-term strategy would beneficial I think. While `display_clade` is probably a little specific as you say, my hope would be for some kind of name that indicates one shouldn't use it in workflows (or use it at your own peril) - so that we can feel more free to adjust it as it suits our needs. I feel like `clade` might be too generic to warn people away from using it - but perhaps something like `clade_combination` or `clade_concat` or similar, might be 'weird sounding' enough that people are more likely to stick to the other options? I'm definitely not overly attached to the current name! On the other hand, people may actually like having something that is guaranteed to always match up with "what Nextstrain is displaying"? (I could see some use to this, though probably narrow.)

I'm also happy to drop `clade_legacy` if we think we may just keep `Nextstrain_clade` as legacy for the foreseeable future. But, if we think we might not do this, then I think having some overlap period where you can switch workflows over (aka, where using both `Nextstrain_clade` works and using `clade_legacy` works) is most useful. Again, would be useful to maybe make a decision around which path we think we'll do - but that's just why I made the PR, to hopefully inspire some discussions/decisions & push forward! 😁

Reply:
Given the thinking and the other PRs in progress, I'd be happy to drop `display_clade` entirely, if we think we can create this internally in `ncov` or otherwise find a way to construct the branch labels we want! I mostly added this because with the complexity of some of the PRs it seemed like the most-least-destructive way to do this, but I now think a better solution (if we can engineer it) would be more stable & clearer.

Yes, happy to wait to introduce the `clade_legacy` until we know exactly what our plan is, as long as we've got some kind of overlap (which we do at the moment, presuming we don't alter this PR to immediately change `Nextstrain-clade`)